### PR TITLE
Khadri 14924 651 pm 

### PR DESCRIPTION
**Pull request by roohishaik128**
Updated Util File

### CODE REVIEW

**Review by SalukulaMahendra — APPROVED** [review]
no conflicts

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 1]
```diff
@@ -6,17 +6,21 @@
 import com.khadri.jdbc.prepare.data.driver.types.DriverTypes;
 import com.khadri.jdbc.prepare.data.student.dao.StudentInsertDao;
 import com.khadri.jdbc.prepare.data.student.processor.StudentDataProcessor;
+import com.khadri.jdbc.prepare.data.supermarket.daos.SuperMarketInsertDao;
+import com.khadri.jdbc.prepare.data.supermarket.processor.SuperMarketDataProcessor;
 
 public class JdbcDataDriver {
 
 	private Scanner scanner;
 	private StudentInsertDao insertDao;
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java]: change name to stdDao or studentInsertDao...

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 2]
```diff
@@ -6,17 +6,21 @@
...
+	private SuperMarketInsertDao superMarketInsertDao;
 	{
 		scanner = new Scanner(System.in);
 		insertDao = new StudentInsertDao();
+		superMarketInsertDao = new SuperMarketInsertDao();
 	}
 
-	public static void main(String[] args) {
+	public static void main(String[] args) throws Exception {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java:36]: handle it try catch.

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 3]
```diff
@@ -28,7 +32,7 @@ public static void main(String[] args) {
 		csvDataDriver.process();
 	}
 
-	private void process() {
+	private void process() throws Exception {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java:48]: dont throws

File: src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java [hunk 4]
```diff
@@ -38,14 +42,29 @@ private void process() {
...
 
 			for (int i = 0; i < recordCount; i++) {
 				stdProcessor.process(dataProcessId);
 			}
 
 			break;
 
+		case 4:
+
+			System.out.println("How many records you want to insert ? : ");
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/driver/JdbcDataDriver.java:94]: add do you..

File: src/main/java/com/khadri/jdbc/prepare/data/supermarket/daos/SuperMarketInsertDao.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Scanner;
+
+import com.khadri.jdbc.prepare.data.student.dao.JdbcConnectionUtil;
+import com.khadri.jdbc.prepare.data.supermarket.dao.model.SuperMarket;
+
+public class SuperMarketInsertDao {
+
+	public static void insertSuperMarketData(SuperMarket market) throws FileNotFoundException {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/supermarket/daos/SuperMarketInsertDao.java]: Which line is throwing FileNotFoundException exception ???

File: src/main/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessor.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		System.out.println("TOTAL_AMT : " + totalAmt);
+		try {
+			for (int i = 0; i < 1; i++) {
+				Thread.sleep(1000);
+				System.out.print("=>");
+			}
+		} catch (Exception e) {
+			System.out.println("never comes");
+		}
+
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessor.java:45]: remove commented lines , if you don't need.

File: src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	@Mock
+	SuperMarketDataProcessor superMarketDataProcessor;
+	
+	@Mock
+	Scanner scanner;
+	
+	@InjectMocks
+	SuperMarketInsertDao superMarketInsertDao;
+	
+	@Mock
```
Comment (khadri2021) [inline src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java:27]: move this mocked variable  to before InjectMocks.

File: src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+	SuperMarket superMarket;
+
+	@Test
+	void testSuperMarketDataProcessor() {
+		fail("Not yet implemented");
+	}
+
+	@Test
+	void testProcess() {
+		Mockito.when(scanner.next()).thenReturn("HANI");
```
Comment (khadri2021) [inline src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java:37]: fix it , write proper test cases.

File: src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+	
+	@InjectMocks
+	SuperMarketInsertDao superMarketInsertDao;
+	
+	@Mock
+	SuperMarket superMarket;
+
+	@Test
+	void testSuperMarketDataProcessor() {
+		fail("Not yet implemented");
```
Comment (khadri2021) [inline src/test/java/com/khadri/jdbc/prepare/data/supermarket/processor/SuperMarketDataProcessorTest.java:32]: If dont need remove it.

File: src/main/java/com/khadri/jdbc/prepare/data/student/dao/JdbcConnectionUtil.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class JdbcConnectionUtil {
+
+	private static Connection con;
+
+	public static Connection getConnection() {
```
Comment (khadri2021) [inline src/main/java/com/khadri/jdbc/prepare/data/student/dao/JdbcConnectionUtil.java:11]: implement the logic to read from db.properties and pass to below DriverManager object creation

**Comment by roohishaik128** [comment]
Updated SuperMarket Module

**Review by thasmiyas39 — APPROVED** [review]

**Review by thasmiyas39 — APPROVED** [review]

**Comment by thasmiyas39** [comment]
approved review

**Pull request merged**